Index: brp-check-suse-1.0/brp-25-symlink
===================================================================
--- brp-check-suse-1.0.orig/brp-25-symlink	2012-03-13 13:52:11.000000000 +0100
+++ brp-check-suse-1.0/brp-25-symlink	2012-07-01 21:37:11.698384089 +0200
@@ -79,6 +79,11 @@ do
 	fi
     done
 
+    if test "$link" = "$link_dest"; then
+      echo "ERROR: $link points to itself (as $orig_link_dest)"
+      exit 1
+    fi
+
     # black list
     case "$link,$link_dest" in
 	*,/var/lib/named*)
