Index: kdecore/services/kservice.cpp
===================================================================
--- kdecore/services/kservice.cpp.orig
+++ kdecore/services/kservice.cpp
@@ -106,6 +106,8 @@ void KServicePrivate::init( const KDeskt
           m_bValid = false;
           return;*/
         m_strType = QString::fromLatin1("Application");
+        if( entryPath.contains( QLatin1String("/konqueror/servicemenus") ))
+            m_strType = QLatin1String("Service"); // workaround for bnc#378338, KDE3 entries there miss Type=
     } else if (m_strType != QLatin1String("Application") && m_strType != QLatin1String("Service")) {
         kWarning(servicesDebugArea()) << "The desktop entry file " << entryPath
                        << " has Type=" << m_strType
