Index: runners/kopete/plasma-runner-kopete.desktop
===================================================================
--- runners/kopete/plasma-runner-kopete.desktop	(revision 1132989)
+++ runners/kopete/plasma-runner-kopete.desktop	(working copy)
@@ -77,4 +77,4 @@
 X-KDE-PluginInfo-Name=kopete_runner
 X-KDE-PluginInfo-Version=0.1
 X-KDE-PluginInfo-License=GPLv2
-X-KDE-PluginInfo-EnabledByDefault=true
+X-KDE-PluginInfo-EnabledByDefault=false
Index: runners/contacts/plasma-runner-contacts.desktop
===================================================================
--- runners/contacts/plasma-runner-contacts.desktop	(revision 1132989)
+++ runners/contacts/plasma-runner-contacts.desktop	(working copy)
@@ -112,5 +112,5 @@
 X-KDE-PluginInfo-Name=kabccontacts
 X-KDE-PluginInfo-Version=1.0
 X-KDE-PluginInfo-License=LGPL
-X-KDE-PluginInfo-EnabledByDefault=true
+X-KDE-PluginInfo-EnabledByDefault=false
 X-Plasma-AdvertiseSingleRunnerQueryMode=true
